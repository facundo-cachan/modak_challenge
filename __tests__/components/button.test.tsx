import { beforeEach, describe, expect, test } from '@jest/globals'
import {
  act,
  fireEvent,
  screen
} from '@testing-library/react-native'

import { Button } from '@atoms'
import { render, withThemeProvider } from '../utils/customRender'

import type { ButtonProps } from '@atoms/button/button-props'

const props: Partial<ButtonProps> = {
  // @ts-ignore
  onPress: jest.fn(),
  title: 'Button title',
  disabled: false,
  loading: false,
  style: {
    backgroundColor: 'red',
  },
};

export const Btn = () => withThemeProvider(<Button {...props} />);
describe('Testing Buttons', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    global.fetch = jest.fn().mockResolvedValue({
      json: () => Promise.resolve([])
    });
    render(<Btn />)
  });
  test('Render', () => {
    expect(screen).toMatchSnapshot()
  });
  test('Visibility title on screen', () => {
    const value = screen.getByText(props.title)
    expect(value).toBeOnTheScreen()
    expect(value).toBeVisible()
    expect(value.props.children).toEqual(props.title)
  })
  test('Find & press button', () => {
    act(() => {
      const pressMe = screen.getByText(props.title);
      expect(pressMe).toBeOnTheScreen();
      fireEvent.press(pressMe);
      act(() => {
        expect(props.onPress).toHaveBeenCalledTimes(1)
      })
    })
  })
  test('Disable & loading', () => {
    render(withThemeProvider(<Button {...props} loading={true} />))
    const value = screen.getByLabelText(props.title)
    expect(value).toBeOnTheScreen()
    expect(value).toBeVisible()
    expect(value.props.children[0].props.icon).toEqual('spinner')
  })
});
