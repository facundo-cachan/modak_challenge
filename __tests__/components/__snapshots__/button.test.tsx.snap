// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Buttons Render 1`] = `
<View
  accessibilityLabel="Button title"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="button"
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "red",
      "borderColor": "#074458",
      "borderRadius": 15,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 30,
      "padding": 10,
    }
  }
>
  <Text
    accessibilityRole="text"
    aria-label="Button title"
    style={
      [
        {
          "flexWrap": "wrap",
        },
        {
          "color": "#115B99",
          "fontFamily": "Arial Hebrew",
          "fontSize": 27,
          "lineHeight": 19.950000000000003,
        },
        [
          {
            "marginLeft": 10,
          },
          undefined,
        ],
      ]
    }
  >
    Button title
  </Text>
</View>
`;
